### PR TITLE
feat: Cloud Gateway operations

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

We needed an operations page that talked about how to provision, upgrade, and delete a CGW.
 

DOCU-3749
### Testing instructions

Preview link: https://deploy-preview-7156--kongdocs.netlify.app/konnect/gateway-manager/provision-cloud-gateway/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

### CODE REVIEW

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
+---
+title: Provision a Dedicated Cloud Gateway
+---
+
+This guide explains how to provision a [Dedicated Cloud Gateway](/konnect/dedicated-cloud-gateways/) and scale the data plane nodes in {{site.konnect_short_name}}.
+
+## Prerequisites
+
+* A network configured in {{site.konnect_short_name}} <!--I feel like a link is missing here, but I can't find a good doc to link to-->
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:9]: I think this is the doc we're waiting for from Wanny.

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+## Provision your fully-managed data plane nodes
+
+{% navtabs %}
+{% navtab UI %}
+1. From [Gateway Manager](https://cloud.konghq.com/gateway-manager) in the navigation menu, click the **New Control Plane** menu and select **{{site.base_gateway}}**.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:15]:
```suggestion
1. From [Gateway Manager](https://cloud.konghq.com/gateway-manager), click the **New Control Plane** button and select **{{site.base_gateway}}**.
```
If I see "the navigation menu", I assume that we're referring to the left side nav for Konnect, but this dropdown is directly on the page.

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+1. In the Create a Control Plane dialog, enter a name for your cloud gateway and click **Dedicated Cloud Instances**.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:17]:
```suggestion
1. Enter a name for your cloud gateway and click **Dedicated Cloud Instances**.
```

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+1. Configure any labels as needed, and click **Next Step**. 
+    
+	You will be redirected the provisioning dialog for your newly created Dedicated Cloud Gateway. The cloud gateway is only created, it still needs to be provisioned and a data plane node needs to be added.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:21]:
```suggestion
	You will be redirected the provisioning dialog for your newly created Dedicated Cloud Gateway. The cloud gateway is created but not yet provisioned, and it still requires a data plane node.
```

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+1. To add a data plane node to your Dedicated Cloud Gateway, do the following in the Create a Data Plane Node dialog:
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:23]:
```suggestion
1. Add a data plane node to your Dedicated Cloud Gateway:
```
The "Create a Data Plane Node dialog" pops up automatically, so we shouldn't need to reorient them.

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+    1. Select one of the following modes to configure your data plane node:
+        * **Autopilot**: This mode allows Kong to automatically scale your instances based on incoming traffic. You can pre-warm your cluster by specifying the number of requests per second. 
+        * **Custom**: This mode allows you to select from three different instance sizes: small, medium, or large.
+
+    1. To configure your cluster, select your provider, region, network, and specify the number of nodes. Repeat this step for as many regions as you need. Adding multiple regions can help with high availability as well as regional failover.
+
+    1. Configure your API access as either public or private, and then click **Create Cluster**. 
+
+Your cloud gateway is now provisioned. You can use it like you would any other {{site.base_gateway}} in {{site.konnect_short_name}}.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:37]:
```suggestion
Your cloud gateway is now provisioned with data plane nodes. You can use it like you would any other {{site.base_gateway}} in {{site.konnect_short_name}}.
```

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+			"port": 443,
+			"protocol": "https"
+			}
+		]
+	},
+	"created_at": "2024-04-04T14:03:46.157Z",
+	"updated_at": "2024-04-04T14:03:46.157Z"
+    }
+    ```
+	Be sure to save the control plane `id` to use in the next step.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:92]:
```suggestion
	Save the control plane `id` to use in the next step. 
```

**Review by lena-larionova — COMMENTED** [review]
First pass review (sorry, saw this was in draft too late, so I hope this still helps.)

File: app/konnect/gateway-manager/provision-cloud-gateway.md (hunk 2 above)
Comment (cloudjumpercat) [reply app/konnect/gateway-manager/provision-cloud-gateway.md:15]: @lena-larionova I was using "menu" in place of drop down here to follow Google's style guide: https://developers.google.com/style/word-list#drop-down. Personally, I agree with you that "menu" is kind of confusing for a drop down, so if we're okay with breaking with Google's guidance there, I can change this to drop down.

Comment (lena-larionova) [reply app/konnect/gateway-manager/provision-cloud-gateway.md:15]: Technically those are buttons with dropdown functionality, so we can just say "button"

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+	--header 'Content-Type: application/json' \
+	--data '{
+		"control_plane_id": "3e62d2c5-45e9-4046-8065-9cd28cb3487d",
+		"version": "3.6",
+		"control_plane_geo": "us",
+		"dataplane_groups": [
+			{
+				"provider": "aws",
+				"region": "ap-northeast-1",
+				"cloud_gateway_network_id": "0e465f2f-4fa3-43ed-8900-bb6210c1e554",
```
Comment (cloudjumpercat) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:109]: @Guaris Was there a way that you found this in the UI? I was trying to and clicking on the network details didn't show me anything. I had to get the value using the API using the network list endpoint.

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+
+## Upgrade your fully-managed data plane nodes
+
+{{site.konnect_short_name}} handles upgrades for you. There's no downtime when upgrading the infrastructure. For more information about how to upgrade your data plane nodes, see [Upgrade a Data Plane Node to a New Version](/konnect/gateway-manager/data-plane-nodes/upgrade/)
+
+## Scale your fully-managed data plane nodes
+
+{% navtabs %}
+{% navtab UI %}
+1. From [Gateway Manager](https://cloud.konghq.com/gateway-manager) in the navigation menu, click the Dedicated Cloud Gateway control plane you want to scale the data plane nodes for.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:181]:
```suggestion
1. From [Gateway Manager](https://cloud.konghq.com/gateway-manager), click the Dedicated Cloud Gateway control plane you want to scale the data plane nodes for.
```
I'm not sure what this instance of "in the navigation menu" is trying to tell me. My first instinct is to look for a list of nodes in a menu somewhere, but they're listed directly on the page.

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+1. Click **Update Cluster**.
+{% endnavtab %}
+{% navtab API %}
+The {{site.konnect_short_name}} API uses [Personal Access Token (PAT)](/konnect/api/#authentication) authentication. You can obtain your PAT from the [personal access token page](https://cloud.konghq.com/global/account/tokens). The PAT must be passed in the `Authorization` header of all requests.
+
+Scale your Dedicated Cloud Gateway data plane nodes by sending a request to the [`/cloud-gateways/configurations` endpoint](link to spec):
+
+```sh
+curl --request PUT \
+--url https://<region>.api.konghq.com/v2/cloud-gateways/configurations \
```
[the remark this answers is not in the document]
Comment (Guaris) [reply app/konnect/gateway-manager/provision-cloud-gateway.md:200]: It's {}

File: app/konnect/gateway-manager/provision-cloud-gateway.md [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+
+{% navtabs %}
+{% navtab UI %}
+1. From [Gateway Manager](https://cloud.konghq.com/gateway-manager), click the Dedicated Cloud Gateway control plane you want to scale the data plane nodes for.
+
+1. Click **Data Plane Nodes** in the navigation menu.
+
+1. From the **Control Plane Actions**, click **Update Cluster Config** and do the following:
+
+    1. To rescale your entire instance, select the most appropriate option based on the requests per second, CPU, and memory from the Custom Configure Mode options.
```
Comment (lena-larionova) [inline app/konnect/gateway-manager/provision-cloud-gateway.md:190]: Is the label really "custom configure mode", or is this a typo?

**Review by lena-larionova — APPROVED** [review]
LGTM - just have one question, and I imagine we won't get the links to the spec until the day of. Can we predict what the links will be, maybe?

File: app/konnect/gateway-manager/provision-cloud-gateway.md (hunk 11 above)
Comment (Guaris) [reply app/konnect/gateway-manager/provision-cloud-gateway.md:190]: The spec link will be [Cloud Gateways API](/konnect/api/cloud-gateways/latest/)

Comment (Guaris) [reply app/konnect/gateway-manager/provision-cloud-gateway.md:190]: The Configure Mode is Custom

image

**Comment by cloudjumpercat** [comment]
Broken links are expected since the specs aren't added yet.

**Pull request merged**